File: backend/app/schemas/payment.py
"""Payment schemas."""

from typing import Optional

from pydantic import BaseModel


class CreateOrderRequest(BaseModel):
    plan: str
    months: int = 1
    payment_method: str = "wechat"  # wechat / alipay


class PayOrderRequest(BaseModel):
    payment_method: str = "wechat"  # wechat / alipay


class UpgradePlanRequest(BaseModel):
    plan: str


class OrderResponse(BaseModel):
    id: str
    order_no: str
    plan: str
    months: int
    amount: float
    currency: str
    status: str
    payment_method: Optional[str] = None
    paid_at: Optional[str] = None
    created_at: str

    class Config:
        from_attributes = True


class PaymentResult(BaseModel):
    order_id: str
    order_no: str
    status: str
    message: str
    payment_url: Optional[str] = None   # 支付宝跳转URL
    code_url: Optional[str] = None      # 微信支付二维码链接
    simulated: bool = False             # 是否为模拟支付（凭证未配置时）

File: frontend/src/api/payment.ts
import client from './client'

export interface PaymentOrder {
  id: string
  order_no: string
  plan: string
  months: number
  amount: number
  currency: string
  status: string
  payment_method?: string | null
  paid_at?: string | null
  created_at: string
}

export interface PaymentResult {
  order_id: string
  order_no: string
  status: string
  message: string
}

export interface CreateOrderRequest {
  plan: string
  months: number
}

export interface PayOrderRequest {
  payment_method: string
}

export interface UpgradePlanRequest {
  plan: string
}

export async function createOrder(data: CreateOrderRequest): Promise<PaymentOrder> {
  const res = await client.post<PaymentOrder>('/payment/orders', data)
  return res.data
}

export async function payOrder(orderId: string, data: PayOrderRequest): Promise<PaymentResult> {
  const res = await client.post<PaymentResult>(`/payment/orders/${orderId}/pay`, data)
  return res.data
}

export async function getOrders(): Promise<PaymentOrder[]> {
  const res = await client.get<PaymentOrder[]>('/payment/orders')
  return res.data
}

export async function upgradePlan(data: UpgradePlanRequest): Promise<{ message: string }> {
  const res = await client.post<{ message: string }>('/payment/upgrade', data)
  return res.data
}
